File: app/src/main/java/com/demo/fluid/activity/WallpaperAdapter.java
package com.demo.fluid.activity;

import android.content.Context;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.bumptech.glide.RequestBuilder;
import com.jakewharton.rxbinding4.view.RxView;
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.TimeUnit;
import kotlin.Unit;
import com.demo.fluid.R;
import com.demo.fluid.activity.home.HomeModel;

public class WallpaperAdapter extends RecyclerView.Adapter<WallpaperAdapter.ViewHolder> {
    private List<HomeModel> _List = new ArrayList();
    private Context context;
    private OnClickListener onClickListener;

    public interface OnClickListener {
        void onClickListener(String str, String str2);
    }

    public WallpaperAdapter(Context context2, OnClickListener onClickListener2) {
        this.context = context2;
        this.onClickListener = onClickListener2;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        return new ViewHolder(LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_wallpaper, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, int i) {
        HomeModel homeModel = this._List.get(i);
        ((RequestBuilder) ((RequestBuilder) Glide.with(this.context).load(getPathThumbnail(homeModel.getTitle())).error(R.drawable.img_thumbnail_default)).placeholder(R.drawable.img_thumbnail_default)).into(viewHolder.image);
        RxView.clicks(viewHolder.rlView).throttleFirst(500, TimeUnit.MILLISECONDS).subscribe(new WallpaperAdapterConsumerCall(this, homeModel));
    }


    public void WallpaperAdapterConsumerCall1(HomeModel homeModel, Unit unit) throws Throwable {
        this.onClickListener.onClickListener(homeModel.getTitle(), homeModel.getAds());
    }

    @Override
    public int getItemCount() {
        return this._List.size();
    }

    private Uri getPathThumbnail(String str) {
        return Uri.parse("file:///android_asset/image/" + str + ".webp");
    }

    public void setData(List<HomeModel> list) {
        this._List = list;
        notifyDataSetChanged();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ImageView image;

        LinearLayout rlView;

        public ViewHolder(View view) {
            super(view);
            this.rlView = (LinearLayout) view.findViewById(R.id.rlView);
            this.image = (ImageView) view.findViewById(R.id.image);

        }
    }
}

File: app/src/main/java/com/demo/fluid/framework/presentation/home/adapter/WallpaperAdapter.kt
package com.demo.fluid.framework.presentation.home.adapter

import android.net.Uri
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.demo.fluid.databinding.ItemHomeBinding
import com.demo.fluid.framework.presentation.model.HomeModel
import pion.tech.fluid_wallpaper.util.loadImage

class WallpaperAdapter :
    ListAdapter<HomeModel, WallpaperAdapter.ViewHolder>(DiffCallback()) {

    interface Listener {
        fun onItemClick(item: HomeModel)
    }

    private var listener: Listener? = null
    fun setListener(listener: Listener) {
        this.listener = listener
    }

    class DiffCallback : DiffUtil.ItemCallback<HomeModel>() {
        override fun areItemsTheSame(
            oldItem: HomeModel,
            newItem: HomeModel
        ): Boolean {
            return oldItem == newItem
        }

        override fun areContentsTheSame(
            oldItem: HomeModel,
            newItem: HomeModel
        ): Boolean {
            return false
        }
    }

    inner class ViewHolder(val binding: ItemHomeBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun bind(item: HomeModel) {
            binding.image.loadImage(getPathThumbnail(item.title))
            binding.root.setOnClickListener {
                listener?.onItemClick(item)
            }
        }
    }

    private fun getPathThumbnail(str: String): Uri {
        return Uri.parse("file:///android_asset/image/$str.webp")
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding = ItemHomeBinding.inflate(inflater, parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        holder.bind(getItem(position))
    }


}